frontend/src/components/SignUp/index.jsx

```diff
@@ -9,47 +11,94 @@ import {
   Paper,
   Typography,
 } from "@mui/material";
-import { useState } from "react";
+import { useEffect, useState } from "react";
+import steps from "@assets/dataStepForm";
+import { cookies } from "../../confCookie";
 import SSignUp from "./style";
 
-const steps = [
-  {
-    label: "Identité",
-    firstField: "Nom",
-    secondeField: "Prénom",
-    thirdField: "Age",
-    type: "default",
-  },
-  {
-    label: "Informations scolaires",
-    firstField: "Nom de l'école",
-    secondeField: "Classe",
-    thirdField: "Option",
-    type: "default",
-  },
-  {
-    label: "Informations de connexion",
-    firstField: "Adresse mail",
-    secondeField: "Mot de passe",
-    thirdField: "Confirmer le mot de passe",
-    type: "password",
-  },
-];
-
 export default function SignUp() {
+  // Variables defined to manage steps in material admin form
   const [activeStep, setActiveStep] = useState(0);
+  // Variables to define dynamically the list of classes from database
+  const [schoolClassList, setSchoolClassList] = useState([]);
+  // Variables and fonction defined to control the form and new authentification
+  const [form, setForm] = useState({
+    email: "",
+    password: "",
+    passwordBis: "",
+    name: "",
+    firstname: "",
+    age: "",
+    schoolOption: "",
+    schoolName: "",
+    schoolClass_id: "",
+  });
 
+  // Function to manage steps in accordeon
   const nextStep = () => {
     setActiveStep((prevActiveStep) => prevActiveStep + 1);
   };
-
   const prevStep = () => {
     setActiveStep((prevActiveStep) => prevActiveStep - 1);
   };
+  // Function to get values from form
+  const hChange = (e) => {
+    const { name, value } = e.target;
+    setForm({ ...form, [name]: value });
+  };
+  // Function to check fields with regex (just mail for the moment)
+  const hCheck = (e, i) => {
+    if (i === "email" && !e.target.value.match(/[\w_-]+@[\w-]+\.[a-z]{2,4}$/i))
+      toast.error(`Votre email n'est pas bon`, {
+        position: "bottom-center",
+        autoClose: 5000,
+        hideProgressBar: false,
+        closeOnClick: true,
+        pauseOnHover: true,
+        draggable: true,
+        progress: undefined,
+      });
+  };
+  // Function to send values in database
+  const hSubmit = (evt) => {
+    evt.preventDefault();
 
-  const reset = () => {
-    setActiveStep(0);
+    axios
+      .post(`${import.meta.env.VITE_BACKEND_URL}/auth/signup`, form)
+      .then(({ data }) => {
+        const { token } = data;
+        cookies.set("token", token);
+        axios.defaults.headers.authorization = `Bearer ${token}`;
+        toast.success(`Félicitations, vous êtes bien inscrit à WIM`, {
+          position: "bottom-center",
```

Comment:
The toast options, you can declare them in the toastcontainer, to avoid repeating them